lib/cocoapods/installer/user_project_integrator/target_integrator.rb

```diff
@@ -147,6 +147,41 @@ def create_or_update_build_phase(native_target, phase_name, phase_class = Xcodep
                 end
               end
           end
+
+          # Updates all target script phases for the current target, including creating or updating, deleting
+          # and re-ordering.
+          #
+          # @return [void]
+          #
+          def create_or_update_user_script_phases(script_phases, native_target)
+            script_phase_names = script_phases.map { |k| k[:name] }
+            # Delete script phases no longer present in the target definition.
+            native_target_script_phases = native_target.shell_script_build_phases.select { |bp| !bp.name.nil? && bp.name.start_with?(USER_BUILD_PHASE_PREFIX) }
+            native_target_script_phases.each do |script_phase|
+              script_phase_name_without_prefix = script_phase.name.sub(USER_BUILD_PHASE_PREFIX, '')
+              unless script_phase_names.include?(script_phase_name_without_prefix)
+                native_target.build_phases.delete(script_phase)
+              end
+            end
+            # Create or update the ones that are expected to be.
+            script_phases.each do |td_script_phase|
+              phase = TargetIntegrator.create_or_update_build_phase(native_target, USER_BUILD_PHASE_PREFIX + td_script_phase[:name])
+              phase.shell_script = td_script_phase[:script]
+              phase.shell_path = td_script_phase[:shell_path] if td_script_phase.key?(:shell_path)
+              phase.input_paths = td_script_phase[:input_files] if td_script_phase.key?(:input_files)
+              phase.output_paths = td_script_phase[:output_files] if td_script_phase.key?(:output_files)
+              phase.show_env_vars_in_log = td_script_phase[:show_env_vars_in_log] ? '1' : '0' if td_script_phase.key?(:show_env_vars_in_log)
+            end
+            # Move script phases to their correct index if the order has changed.
```

Comment:
What does `correct index` mean and is there an interdependency with the phases coming from the podspec?

Reply:
@paulb777 this existed for the Podfile script phases and it will be updated to support `:execution_placement` DSL (PR will be open in Core after this lands). This logic was not added as part of this PR.

The current logic basically re-arranges script phases to match their order within a Podfile.

Given
```ruby
  target 'MyApp' do
    script_phase :name => 'Hello World', :script => 'echo "Hello World"'
    script_phase :name => 'Hello World 2', :script => 'echo "Hello World 2"'
  end
```

If the user goes into the Podfile and switches the order then this code would automatically switch the order as well.

I am planning on removing/updating this before 1.4.0.